src/nupic/research/spatial_pooler.py

```diff
@@ -584,89 +559,107 @@ def setSynPermConnected(self, synPermConnected):
     connected"""
     self._synPermConnected = synPermConnected
 
-
-  def getMinPctOverlapDutyCycles(self):
-    """Returns the minimum tolerated overlaps, given as percent of
-    neighbors overlap score"""
-    return self._minPctOverlapDutyCycles
+# Boosting related get/set() methods
+  def getMaxBoost(self):
+    """Returns the maximum boost value"""
+    return self.boosting.getMaxBoost()
 
 
-  def setMinPctOverlapDutyCycles(self, minPctOverlapDutyCycles):
-    """Sets the minimum tolerated activity duty cycle, given as percent of
-    neighbors' activity duty cycle"""
-    self._minPctOverlapDutyCycles = minPctOverlapDutyCycles
+  def setMaxBoost(self, maxBoost):
+    """Sets the maximum boost value"""
+    self.boosting.setMaxBoost(maxBoost)
 
 
-  def getMinPctActiveDutyCycles(self):
-    """Returns the minimum tolerated activity duty cycle, given as percent of
-    neighbors' activity duty cycle"""
-    return self._minPctActiveDutyCycles
+  def getBoostFactors(self):
+    """Returns the boost factors for all columns."""
+    return self.boosting.getBoostFactors()
 
 
-  def setMinPctActiveDutyCycles(self, minPctActiveDutyCycles):
-    """Sets the minimum tolerated activity duty, given as percent of
-    neighbors' activity duty cycle"""
-    self._minPctActiveDutyCycles = minPctActiveDutyCycles
+  def setBoostFactors(self, boostFactors):
+    """Sets the boost factors for all columns. 'boostFactors' size must match
+    the number of columns"""
+    self.boosting.setBoostFactors(boostFactors)
 
 
-  def getBoostFactors(self, boostFactors):
-    """Returns the boost factors for all columns. 'boostFactors' size must
-    match the number of columns"""
-    boostFactors[:] = self._boostFactors[:]
```

Comment:
@scottpurdy I've changed the getters to `pass-by-return-value`, instead of "pass by param", as I find it clearer and I think I remember from somewhere that we want to prefer that approach. Is it ok? Also, it would require changes in c++/swig, so I'm not sure if for this PR, or for a separate one?